homeassistant/components/google_assistant/helpers.py

```diff
@@ -141,18 +161,34 @@ def async_schedule_google_sync(self, agent_user_id: str):
             self.hass, SYNC_DELAY, _schedule_callback
         )
 
+    @callback
+    def async_schedule_google_sync_all(self):
+        """Schedule a sync for all registered agents."""
+        for agent_user_id in self._store.agent_user_ids:
+            self.async_schedule_google_sync(agent_user_id)
+
     async def _async_request_sync_devices(self, agent_user_id: str) -> int:
         """Trigger a sync with Google.
 
         Return value is the HTTP status code of the sync request.
         """
         raise NotImplementedError
 
-    async def async_deactivate_report_state(self):
+    async def async_connect_agent_user(self, agent_user_id: str):
+        """Add an synced and known agent_user_id.
+
+        Called when a completed sync response have been sent to Google.
+        """
+        self._store.agent_user_ids.add(agent_user_id)
+        self._store.schedule_save()
```

Comment:
this will do a lot of writes. Delegate this to the store. Inside the store, check if it is already known agent_user_id, and if so, do nothing. Otherwise, write.